src/geometry/triangle.rs

```diff
@@ -97,11 +97,11 @@ impl Intersectable for Triangle {
         let pvec = ray.direction.cross(&self.ac);
         let det = self.ab.dot(&pvec);
 
-        if cull && det < EPSILON {
+        if cull && det < 1e-7 {
```

Comment:
Is there any reason you replaced `EPSILON` here instead of changing the value of `EPSILON`?

Reply:
Yes, 1e-6 is too large and caused certain rays to be counted as misses when in fact they were hits. However, math::EPSILON is used elsewhere and has an appropriate value for those usecases.